pennylane/ops/functions/equal.py

```diff
@@ -658,41 +658,54 @@ def _equal_measurements(
     """Determine whether two MeasurementProcess objects are equal"""
 
     if op1.obs is not None and op2.obs is not None:
-        return equal(
+        dispatch_result = _equal(
             op1.obs,
             op2.obs,
             check_interface=check_interface,
             check_trainability=check_trainability,
             rtol=rtol,
             atol=atol,
         )
+        if isinstance(dispatch_result, str):
+            return dispatch_result
```

Comment:
This would indicate that the observables are not equal, but we would also want to add some context this message to indicate that this message is about measurements.

How about something like:
```suggestion
            return f"op1 and op2 have different observables because {dispatch_result}"
```